core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -308,6 +309,12 @@ public UpdateSchema moveAfter(String name, String afterName) {
     return this;
   }
 
+  @Override
+  public UpdateSchema unionWithByFieldName(Schema newSchema) {
+    UnionByNameVisitor.visit(this, schema, newSchema);
```

Comment:
@rdblue you're right, the implementation, while a bit more complex than previous one feels less complicated.